azure-storage-file/azure/storage/file/fileservice.py

```diff
@@ -1259,6 +1263,187 @@ def _list_directories_and_files(self, share_name, directory_name=None,
+        operation_context = _OperationContext(location_lock=True)
+        args = (share_name, directory_name, file_name)
+        kwargs = {'marker': marker, 'max_results': max_results, 'timeout': timeout, 'recursive': recursive,
+                  '_context': operation_context, 'snapshot': snapshot}
+
+        resp = self._list_handles(*args, **kwargs)
+
+        return ListGenerator(resp, self._list_handles, args, kwargs)
+
+    def _list_handles(self, share_name, directory_name=None, file_name=None, recursive=None,
+                      marker=None, max_results=None, timeout=None, _context=None, snapshot=None):
+        """
+        Returns a list of the directories and files under the specified share.
+
+        :param str share_name:
+            Name of existing share.
+        :param str directory_name:
+            The path to the directory.
+        :param str file_name:
+            Name of existing file.
+        :param bool recursive:
+            Boolean that specifies if operation should apply to the directory specified in the URI,
+            its files, its subdirectories and their files.
+        :param str marker:
+            An opaque continuation token. This value can be retrieved from the
+            next_marker field of a previous generator object if max_results was
+            specified and that generator has finished enumerating results. If
+            specified, this generator will begin returning results from the point
+            where the previous generator stopped.
+        :param int max_results:
+            Specifies the maximum number of handles to return,
+            including all directory elements. If the request does not specify
+            max_results or specifies a value greater than 5,000, the server will
+            return up to 5,000 items. Setting max_results to a value less than
+            or equal to zero results in error response code 400 (Bad Request).
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :param str snapshot:
+            A string that represents the snapshot version, if applicable.
+        """
+        _validate_not_none('share_name', share_name)
+        request = HTTPRequest()
+        request.method = 'GET'
+        request.host_locations = self._get_host_locations()
+        request.path = _get_path(share_name, directory_name, file_name)
+        request.query = {
+            'comp': 'listhandles',
+            'marker': _to_str(marker),
+            'maxresults': _int_to_str(max_results),
+            'timeout': _int_to_str(timeout),
+            'sharesnapshot': _to_str(snapshot)
+        }
+        request.headers = {
+            'x-ms-recursive': _to_str(recursive)
+        }
+
+        return self._perform_request(request, _convert_xml_to_handles,
+                                     operation_context=_context)
+
+    def close_handles(self, share_name, directory_name=None, file_name=None, recursive=None,
```

Comment:
This feels confusing to use.  If it's not too complicated, could we have a sample (or test) that shows how to make sense of the output of the generator?  Something like:

totalHandlesClosed = 0
for handleCountThisIteration in close_handles(...):
    totalHandlesClosed += handleCountThisIteration
print(totalHandlesClosed)

Otherwise, it's not super clear to me why there's an iteration going on at all

Reply:
Ahh ok, you do explain it in the comments.  Still though, might be good to show it.  Do you know if I have to follow continuation tokens if i'm only closing a single handle?

Reply:
Sure, I've modified the existing test to use the generator directly. 
No it's most likely not going to give token for single handle (with the id specified). But I assume it's possible if "*" is given and many iterations happen.